### PR TITLE
utils/shared_audits.rb: check version :all for non-cask formulae in the prerelease allowlist

### PR DESCRIPTION
**Pull request by Moisan**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?
- [x] Have you successfully run `brew man` locally and committed any changes?

-----

All `gron` releases are pre-releases: https://github.com/tomnomnom/gron/releases

While adding this formula to the prerelease allowlist, I spotted that `:all` version was only used for casks.

### CODE REVIEW

File: Library/Homebrew/utils/shared_audits.rb [hunk 1]
```diff
@@ -59,7 +60,8 @@ def github_release(user, repo, tag, formula: nil, cask: nil)
     end
 
     if release["prerelease"]
-      return if formula && GITHUB_PRERELEASE_ALLOWLIST[formula.name] == formula.version
+      return if formula && (GITHUB_PRERELEASE_ALLOWLIST[formula.name] == formula.version ||
+                            GITHUB_PRERELEASE_ALLOWLIST[formula.name] == :all)
```
Comment (reitermarkus) [inline Library/Homebrew/utils/shared_audits.rb:64]: This should probably also fail if it's not a pre-release anymore but still in the `GITHUB_PRERELEASE_ALLOWLIST` like above.

File: Library/Homebrew/utils/shared_audits.rb [hunk 2]
```diff
@@ -46,6 +46,7 @@ def github_release_data(user, repo, tag)
     "telegram-cli"     => "1.3.1",
     "toggl-track"      => :all,
     "volta"            => "0.8.6",
+    "gron"             => :all,
```
Comment (miccal) [inline Library/Homebrew/utils/shared_audits.rb:49]: Sorry to be picky, but would you mind putting this in alphabetical order? I regularly add Casks to this list, and having them in alphabetical order really helps. Thanks.

Comment (Moisan) [reply Library/Homebrew/utils/shared_audits.rb:49]: Sure, no problem!

Comment (miccal) [reply Library/Homebrew/utils/shared_audits.rb]: Thank you @Moisan.

**Comment by MikeMcQuaid** [comment]
@Moisan Can you rebase this and shout if/when it's ready? Thanks!

**Comment by Rylan12** [comment]
I don't think this is needed anymore. The logic was modified in https://github.com/Homebrew/brew/pull/10087 and the lists were migrated to their respective taps. Formulae should not have the ability to specify `all` by adding this line to `$(brew --repo homebrew/core)/audit_exceptions/github_prerelease_allowlist.json`:

```json
{
  ...
  "gron": "all",
  ...
}
```

**Comment by Moisan** [comment]
Perfect, I'll close this PR and open one to add gron to the pre-release list in homebrew-core.

**Pull request closed without merging**